services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/ScanEnhancedRequest.java

```diff
@@ -203,6 +231,44 @@ public Builder limit(Integer limit) {
             return this;
         }
 
+        /**
+         * For a parallel Scan request, Segment identifies an individual segment to be scanned by an application worker.
+         * <p>
+         * <b>Note:</b>Segment IDs are zero-based, so the first segment is always 0. For example, if you want to use four
+         * application threads to scan a table or an index, then the first thread specifies a Segment value of 0, the second
+         * thread specifies 1, and so on.
+         *
+         * The value of {@link Page#lastEvaluatedKey()} returned from a parallel Scan request must be used as
+         * {@link #exclusiveStartKey} with the same segment ID in a subsequent Scan operation.
```

Comment:
It depends on what return type. If the user uses SdkIterable<Record>, then it takes care of pagination. If the user uses Iterator<Page<Record>> , then the pagination tokens needs to be set for next call. 

This doc explains that within each 'segment', we need to make paginated calls. This doc is copied from underlying client [ScanRequest](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/dynamodb/model/ScanRequest.html#segment-- ) model.

Reply:
In the case of the DynamoDBEnhancedClient I believe it's always handled by the SDK. If that belief is correct, then I'd be in favor of removing this line to prevent confusion.

Reply:
Agreed. I removed it in next commit.